lib/pure/asyncfutures.nim

```diff
@@ -30,7 +33,14 @@ type
 when not defined(release):
   var currentID = 0
 
-proc callSoon*(cbproc: proc ()) {.gcsafe.}
+var callSoonProc* {.threadvar.}: (proc(cbproc: proc ()) {.gcsafe.})
+
+proc callSoon*(cbproc: proc ()) =
+  if callSoonProc == nil:
+    # Loop not initialized yet. Call the function directly to allow setup code to use futures.
```

Comment:
Yes, it has. But I think it's unlikely to hit this bug before initializing the dispatcher. If the dispatcher is not initialized in some other place, `callSoon` will anyway never be called.

I would like to avoid making `asyncfutures` depend on `asyncdispatch`. My "ultimate goal" is making event loops pluggable, in a similar manner to Pythons asyncio.